src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -2328,7 +2328,7 @@ public void BinaryLogContainsImportedFiles()
             </Project>
             ");
 
-            string binLogLocation = _env.DefaultTestDirectory.Path;
+            string binLogLocation = FileUtilities.TempFileDirectory;
```

Comment:
This is a change I don't expect, what's the motivation?

Reply:
The test was failing, because archive file wasn't found on that location. We agreed that we will change the archive location to not interfere with the file watchers (which was motivation of keeping data in memory).

Reply:
Ah, thanks! I think this is actually a bug in the new implementation.

When in `ProjectImports=ZipFile` mode, we expect *two* output files:

1. the binlog itself, with no embedded  content files
2. a zip with the extension `.ProjectImports.zip` with the content files

We want that zip to be next to the .binlog _when it is an output of the build_. But when it's an internal implementation detail to avoid memory, we want it somewhere else. Looks like the current code moves it unconditionally, which isn't right.

I see two options:

1. tweak the location based on mode (the old `createFile` argument to the `ProjectImportsCollector` ctor)
2. _move_ the file to the final location instead of deleting it, based on mode

I prefer the former.